Task4_AWSCloud Formation.yaml

```diff
@@ -0,0 +1,161 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  CloudFormation template For Networking and Subnets to create a VPC with public and private subnets in multiple availability zones.
+
+Parameters:
+  VpcCidr:
+    Type: String
+    Default: 172.16.0.0/16
+    Description: CIDR block for the VPC
+  PublicSubnetCidr1:
+    Type: String
+    Default: 172.16.1.0/24
+    Description: CIDR block for Public Subnet 1 AZ1
+  PublicSubnetCidr2:
+    Type: String
+    Default: 172.16.2.0/24
+    Description: CIDR block for Public Subnet 2 AZ1
+  PrivateSubnetCidr1:
+    Type: String
+    Default: 172.16.3.0/24
+    Description: CIDR block for Private Subnet 1 AZ2
+  PrivateSubnetCidr2:
+    Type: String
+    Default: 172.16.4.0/24
+    Description: CIDR block for Private Subnet 2 AZ2
+
+Resources:
+ #VPC && InternetGateway.
+  #---- VPC Creation.
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VpcCidr
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: DevVPC
+  #---- InternetGateway Creation.
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: MyInternetGateway
+  #---- AttachGateway To VPC.
+  AttachGateway:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref VPC
+      InternetGatewayId: !Ref InternetGateway
+  
+ #RouteTables
+  #---- Public RouteTable Creation (Info:Any component will assassin here will send/receive requests to/from anywhere).
+  PublicRouteTable:
+    Type: AWS::EC2::RouteTable

```

Comment:
RouteTable in the risk of out of order creation, it might be created before the referenced vpc. In order to solve it you can use `DependOn` Attribute which can ensure a correct components creation